flexmeasures/cli/data_delete.py

```diff
@@ -280,23 +297,35 @@ def delete_unchanged_beliefs(
             include_positive_horizons=False,
         )
         unchanged_queries.append(q_unchanged_measurements)
-        num_measurements_up_for_deletion = q_unchanged_measurements.count()
+        num_measurements_up_for_deletion = db.session.scalar(
+            select(func.count()).select_from(q_unchanged_measurements)
+        )
 
     num_beliefs_up_for_deletion = (
         num_forecasts_up_for_deletion + num_measurements_up_for_deletion
     )
     prompt = f"Delete {num_beliefs_up_for_deletion} unchanged beliefs ({num_measurements_up_for_deletion} measurements and {num_forecasts_up_for_deletion} forecasts) out of {num_beliefs_before} beliefs?"
     click.confirm(prompt, abort=True)
 
-    beliefs_up_for_deletion = list(chain(*[q.all() for q in unchanged_queries]))
+    beliefs_up_for_deletion = list(
+        chain(*[db.session.scalars(q).all() for q in unchanged_queries])
+    )
     batch_size = 10000
     for i, b in enumerate(beliefs_up_for_deletion, start=1):
         if i % batch_size == 0 or i == num_beliefs_up_for_deletion:
             click.echo(f"{i} beliefs processed ...")
-        db.session.delete(b)
+        db.session.execute(
+            delete(TimedBelief).filter_by(
+                belief_horizon=b.belief_horizon,
+                event_start=b.event_start,
+                sensor_id=b.sensor_id,
+                source_id=b.source_id,
+                cumulative_probability=b.cumulative_probability,
+            )
+        )
```

Comment:
I noticed that this is significantly slower than using `db.session.delete(b)`. The `session.delete(b)` syntax still seems to work and I didn't find mention of it being deprecated. As a reference: deleting 8200 beliefs took me 2.5 minutes with the `delete(b)` syntax, and 9 minutes with the `delete(TimedBelief).filter(...)` syntax. Although I consider both to be unreasonably slow, I'd rather use the `session.delete(b)` syntax if it's still valid.

Reply:
Can we generate the DELETE sql query? I believe we might be able to uncover a structural problem here, just as with the SELECT.

Reply:
Just wanted to note that the `db.session.delete(b)` syntax leads to most time being taken on the `db.session.commit`, so I had a hunch it has to do with reindexing the table.

Reply:
I think, we should revert this change only, while the rest of the `delete` statement should remain the same because for them  we have to update the `select` statements too.

Reply:
Agreed, but I think there are 2 CLI commands that deal with deleting beliefs: one deletes NaN beliefs, the other deletes unchanged beliefs. Let's revert these two.